### PR TITLE
PI instance

### PR DESCRIPTION
**Pull request by sonkatomi**

### CODE REVIEW

File: internal/providers/terraform/ibm/testdata/pi_instance_test/pi_instance_test.golden [hunk 1]
```diff
@@ -0,0 +1,21 @@
+
+ Name                           Monthly Qty  Unit      Monthly Cost 
+                                                                    
+ ibm_pi_instance.test-instance
```
Comment (balazs-marjan) [inline internal/providers/terraform/ibm/testdata/pi_instance_test/pi_instance_test.golden:4]: More test cases please

File: internal/resources/ibm/pi_instance.go [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+// PiInstance struct represents <TODO: cloud service short description>.
+//
+// <TODO: Add any important information about the resource and links to the
+// pricing pages or documentation that might be useful to developers in the future, e.g:>
+//
+// Resource information: https://cloud.ibm.com/<PATH/TO/RESOURCE>/
```
Comment (balazs-marjan) [inline internal/resources/ibm/pi_instance.go:18]: Add information about the resources

**Review by balazs-marjan — CHANGES_REQUESTED** [review]

File: internal/providers/terraform/ibm/testdata/pi_instance_test/pi_instance_test.golden [hunk 3]
```diff
@@ -0,0 +1,36 @@
+
+ Name                                            Monthly Qty  Unit      Monthly Cost 
+                                                                                     
+ ibm_pi_instance.aix-shared-s922-instance
```
Comment (balazs-marjan) [inline internal/providers/terraform/ibm/testdata/pi_instance_test/pi_instance_test.golden:4]: As discussed can you:
- change the usage up to take a montly instance hours param?
- make test cases that don't have the instance hours params supplied?

File: internal/providers/terraform/ibm/pi_instance.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		SystemType:             systemType,
+		ProcessorMode:          processorMode,
+		Cpus:                   cpus,
+		Memory:                 memory,
+		StorageType:            storageType,
+		OperatingSystem:        os,
+		LegacyIBMiImageVersion: isLegacyIBMiImageVersion,
+		NetweaverImage:         isNetweaverImage,
+	}
+	r.PopulateUsage(u)
```
Comment (balazs-marjan) [inline internal/providers/terraform/ibm/pi_instance.go:109]: Can you add the catalog metadata here as with all others resources?

**Review by balazs-marjan — CHANGES_REQUESTED** [review]
Can you also add the pi_instance resource to the `infracost_usage_example.yml` with comments and so on?

File: internal/providers/terraform/ibm/testdata/pi_instance_test/pi_instance_test.golden (hunk 3 above)
Comment (sonkatomi) [reply internal/providers/terraform/ibm/testdata/pi_instance_test/pi_instance_test.golden:4]: noted

**Review by balazs-marjan — CHANGES_REQUESTED** [review]

**Review by balazs-marjan — APPROVED** [review]

**Pull request merged**